### DISCUSSION TITLE
KTL Version Control

### DISCUSSION THREAD

**Discussion by CSWinnall** [cortexrd/Knack-Toolkit-Library#241] [General]
We use the KTL throughout our app so we cannot have the KTL down or turned off for any reason.  To protect against this we  stay 1 or 2 versions behind.

Norm releases updates fairly frequently with amazing features and while this method is occasionally frustrating because we can't have those features instantly, we find that they are worth the wait.

In the JavaScript pane, in the loader, you will see this

https://github.com/cortexrd/Knack-Toolkit-Library/assets/10554243/530fd773-4e61-4a5f-b2da-6397e0499a3b

Where you see 0.22.20 you can replace this with any version that Norm has released.  This will offer some protection from very rare errors that miss Norms vigorous testing.  You can then test the latest releases on a test app. If you are happy with your testing you can then advance the version number on your live app to one you are happy with.

As I said before the downside to this is your users could miss out on the newest features for a week or 2 but in return you have control of the changes to your app.

If you have any queries about this then please let us know.

**Comment by chprojectsupport** [comment]
Sound advice ☑️
Given the issues today of all forms not submitting I recommended to my only client using KTL to stay a step behind by adding the previous release to the loader. 
Makes sense and worth the wait for new features balanced against unforeseen issues. 👍
